08-VendoringDependenciesAsSubmodules.adoc

```diff
@@ -286,17 +260,21 @@ From the submodule initialize and update output:
 
 `git submodule update` can take some parameters to customize its behavior:
 
 * The `--recursive` flag, which will run `git submodule update --init` inside each of the submodules directories too. This is useful when there are nested submodules inside submodules.
-* The `--no-fetch` flag will attempt to update the submodule without running `git fetch`. This will only update the submodule to a later revision if this has already been fetched. This is useful if you want to fetch the changes to a submodule now and then update and test this update at a later point.
-* The `--force` (or `-f`) flag can be passed to update the submodules to the latest revision by running the equivalent of `git checkout --force`--to discard any uncommitted changes made to the submodule.
+* The `--force` (or `-f`) flag can be passed to update the submodules to the commit recorded in the superproject by running the equivalent of `git checkout --force`--to discard any uncommitted changes made to the submodule.
 * The `--depth` is passed to the `git clone` of the submodule to allow creating a shallow clone with only the requested number of revisions within it. This can be used to shrink the size of the submodule on disk.
 
 `git clone` can also take a `--recurse-submodules` (or `--recursive`) flag to automatically run `git submodule update --init` on any submodules within the repository. Typically if you're cloning a repository you know contains submodules, then you'll use `git clone --recursive-submodules` to clone it and all the necessary submodules (and the submodules of the submodules, if they exist).
 
+Sometimes, you may want to update the submodule to the latest upstream revision to incorporate any changes that were made in the upstream, submodule repository. 
+In that case, `git submodule update` has other useful parameters:
+
+* The `--remote` flag would fetch and checkout the latest upstream revision in the local submodule repository. This would then require another commit and push to update this on the `GitInPracticeRedux` remote repository. This should only be done after testing that the changes made to the `GitInPracticeReduxSubmodule` repository remain compatible with the `GitInPracticeRedux` project.
+* The `--no-fetch` flag will attempt to update the submodule without running `git fetch`. This will only update the submodule to a later revision if this has already been fetched. This is useful if you want to fetch the changes to a submodule now and then update and test this update at a later point.
```

Comment:
```suggestion
* The `--remote` flag will fetch and checkout the latest upstream revision in the local submodule repository. This would then require another commit and push to update this on the `GitInPracticeRedux` remote repository.
* The `--no-fetch` flag will attempt to update the submodule without running `git fetch`. This will only update the submodule to a later revision if this has already been fetched. This is useful if you want to fetch the changes to a submodule now and then update and test this update at a later point.
```

Reply:
About the deletion of "This should only be done after testing..": 
I actually copied that from https://github.com/MikeMcQuaid/GitInPractice/pull/21/files#diff-331074c6cb7d301355846137de0cff49L289 (just to make sure you noticed; it's not a new sentence I wrote). I thought it made sense to warn readers that submodules updates should be tested before commiting it, to make sure i.e. the API did not change, etc., that's why I copied it here.

Reply:
Also, while I'm re-reading it now it think this addition could also make sense:
```suggestion
* The `--remote` flag would fetch and checkout the latest upstream revision in the local submodule repository. This would then require another commit to update this on the local `GitInPracticeRedux` repository, and a push to update this on the `GitInPracticeRedux` remote repository. This should only be done after testing that the changes made to the `GitInPracticeReduxSubmodule` repository remain compatible with the `GitInPracticeRedux` project.
* The `--no-fetch` flag will attempt to update the submodule without running `git fetch`. This will only update the submodule to a later revision if this has already been fetched. This is useful if you want to fetch the changes to a submodule now and then update and test this update at a later point.
```

Reply:
Whoops, re-added. Good idea, I've adjusted my suggestion based on yours.